Makefile

```diff
@@ -2074,6 +2078,15 @@ uid.o: \
 compile uid.c uidgid.h subfd.h substdio.h exit.h
 	./compile uid.c
 
+unittest_stralloc: \
```

Comment:
Please put these into a subdirectory: it will cause much less jitter in the main Makefile, causing less collisions with other patches and PRs.

Reply:
I'm in favor of avoiding those collisions, but I'm not sure how best to handle the dependencies in a separate dir and/or Makefile. If there's a way you might like, would you append a commit and push to this branch so I can see what you mean?